File: marc/examples/hamiltonian-cycle/temperature/GeometricTemperature.cpp
#include <cmath>
#include "GeometricTemperature.h"

double GeometricTemperature::temperature(const unsigned int step) const {
    return pow(alpha, step-1)*T0;
}

GeometricTemperature::GeometricTemperature(unsigned int T0, const double alpha) : alpha(alpha), T0(T0) { }

File: jmarc/examples/NQueens/GeometricTemperature.java
import jmarc.solving.MARCTemperature;

public class GeometricTemperature extends MARCTemperature {

    private double initialTemp;
    private double alpha;

    public GeometricTemperature(double initialTemp, double alpha) {
        super();
        this.initialTemp = initialTemp;
        this.alpha = alpha;
    }

    @Override
    public double temperature(int step) {
        // Formula: T_corrente = T_iniziale * (alpha ^ numero_step)
        return initialTemp * Math.pow(alpha, step);
    }
}
